snapshot/restore/restore.go

```diff
@@ -20,6 +20,7 @@ var log = logging.GetContextLoggerFunc("restore")
 type Output interface {
 	Parallelizable() bool
 	BeginDirectory(ctx context.Context, relativePath string, e fs.Directory) error
+	WriteShallowDirectory(ctx context.Context, relativePath string, e fs.Directory) error
```

Comment:
can this be `WriteDirEntry()` and take DirEntry directly, IOW the caller would be responsible for figuring out how to get the DirEntry?

Reply:
Done but I'll leave it to you to decide if it's actually better this way. I'm not convinced: `WriteFile` and `WirteDirEntry` are more divergent than they were before and maybe harder to follow.

Reply:
Done in 0a2ce50